src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/TestHelper.java

```diff
@@ -77,6 +77,15 @@ public void uploadZipFile(
         blockBlobReference.uploadFromByteArray(zipFile, 0, zipFile.length);
     }
 
+    public void uploadAndLeaseZipFile(
+        CloudBlobContainer container, String srcZipFilename, String destZipFilename
+    ) throws Exception {
+        byte[] zipFile = Resources.toByteArray(Resources.getResource(srcZipFilename));
+        CloudBlockBlob blockBlobReference = container.getBlockBlobReference(destZipFilename);
+        blockBlobReference.uploadFromByteArray(zipFile, 0, zipFile.length);
+        blockBlobReference.acquireLease();
```

Comment:
This is not going to work consistently . You are uploading the file and then acquiring the lease which sometime may work if the process is able to acquire the lock else it will fail as other processes(jobs) may be running.

You will need to do something as below

```

 CloudBlockBlob blockBlobReference = container.getBlockBlobReference(destZipFilename);
        blockBlobReference.acquireLease(15,"1234");
        blockBlobReference.uploadFromByteArray(zipFile, 0, zipFile.length, AccessCondition.generateLeaseCondition("1234"),null,null);
....
```

Reply:
@nitinprabhu I agree with you on the race condition and I tried already what you suggest but I got an error as the blob had not been created when yet when requesting the lease. For now this is better than what we currently have.

Reply:
ok makes sense to acquire lease after blob is uploaded. Have you tried below ?

```

CloudBlockBlob blockBlobReference = container.getBlockBlobReference(destZipFilename);
        blockBlobReference.uploadFromByteArray(zipFile, 0, zipFile.length, AccessCondition.generateLeaseCondition("1234"),null,null);
        blockBlobReference.acquireLease(15,"1234");```